### ISSUE TITLE
[bug] Minimal project fails to compile - using CodeSandbox code from Readme...

### ISSUE THREAD

**Issue by RichHorrocks** [rainbow-me/rainbowkit#443]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

0.1.0

### wagmi Version

0.3.2

### Current Behavior

I downloaded the .zip file from the `create-react-app`-based example, linked from the Readme's _Try It Out_ section. (This -> https://codesandbox.io/s/dn3rho)

I then `yarn install`'ed and started the project, but hit the issue below. I have made no changes to the code that I downloaded from CodeSandbox.

```
Failed to compile.

./node_modules/react-query/lib/reactjs/useSyncExternalStore.mjs
Can't reexport the named export 'useSyncExternalStore' from non EcmaScript module (only default export is available)
```

### Expected Behavior

The example project should run.

### Steps To Reproduce

As described above.

**Comment by RichHorrocks** [comment]
It appears to be a bug in `react-query` -> https://github.com/tannerlinsley/react-query/issues/3637.

Any idea if this can be worked around? (No idea why the code runs in CodeSandbox itself... )

**Comment by jxom** [comment]
Can you try bump `react-scripts` to `^5`?

Re: https://github.com/tmm/wagmi/issues/398#issuecomment-1133659610

**Comment by RichHorrocks** [comment]
Thanks for the reply, and for pointing to that Wagmi issue.

I _can_ bump `react-scripts`, but then I get lots of Webpack v5 errors. I think most of them are related to WalletConnect, and there seems to be some discussion going on here: https://github.com/rainbow-me/rainbowkit/issues/420#issuecomment-1138861792

```
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.
```

I'll have a play around with `react-app-rewired` and see if I can configure some defaults for the complaining packages.

**Comment by TkDodo** [comment]
we've fixed the upstream issue, sorry for that. see:

https://github.com/tannerlinsley/react-query/releases/tag/v4.0.0-beta.17

**Comment by jxom** [comment]
`wagmi@0.4.5` is now released which upgrades `react-query` to `4.0.0-beta.19` and hopefully fixes this issue.

Please upgrade `wagmi` and try again 🙏.

**Comment by RichHorrocks** [comment]
Yep, perfect - all working as it should now.

Thanks for the speedy turnaround on this!
